### ISSUE TITLE
Scenario class: handling mechanism for un-updated recovered cases data

### ISSUE THREAD

**Issue by Inglezos** [lisphilar/covid19-sir#269]
## Summary of new feature - Suggestions
It would be very useful to implement a mechanism that handles cases when reliable recovered data cannot be provided, for example in situations where recovered data are:
- frozen for a long time
- not updated every day
- unknown/not reported

See also #268 issue.

I also suspect that the source **recovered** data for **UK** and **Netherlands** are **not correct** (they seem too little). In the `worldometers.info` they do not exist at all (N/A).

## Solution
- In such cases, the system maybe should **discard such invalid data** and not use them in the confirmed/deaths calculations/predictions. Perhaps by not including the recovered data in the **cost function** for the prediction.
- Alternatively, a **generator function** perhaps should be developed to provide such data by approximation.
- Or we could use the **theoretical data** of the model for the expected recovered cases, which correspond to the specific current parameters.

**Comment by lisphilar** [comment]
Just a note for a generator function, parameter estimation:
- moving average of actual records
- spline interpolation
- perform time series analysis and find trend with fbprophet

**Comment by Inglezos** [comment]
I think that this issue has to be handled first and soon, because it creates many problems both in the trends analysis and the estimations. Many countries have either frozen recovered cases or do not update them regularly.

**Comment by lisphilar** [comment]
We will create `Scenario.complement(method, model=None, target="Recovered")` method. 

1. We do not have detection system of irreguration in the current version.
You suggested three solutions. How do you write Python codes for them with the output of `Scenario.records()`?
- frozen for a long time
- not updated every day
- unknown/not reported

2. How to complement R:

> We could use the estimated parameters of the phase at which the recovered cases begin to show irregularities. 

When we know irregularities starts at 3rd phase, we can fix `sigma` in 3rd phase as follows.
```Python
snl = cs.Scenario(...)
snl.trend()
# Parameter estimation in 0th, 1st, 2nd phase
snl.estimate(cs.SIRF, phases=["0th", "1st", "2nd"])
# Get sigma value in 2nd phase
sigma_2nd = snl.get("sigma", phase="2nd")
# Use 2nd phase's sigma value in 3rd phase
snl.estimate(cs.SIRF, phases=["3rd"])
```

**Comment by Inglezos** [comment]
We can easily detect the irregularities if we simply count the records of consecutive non-increasing values and set a tolerance threshold for that. If for example a country has stuck/non-increasing recovered cases for consecutive let's say 3 days, then the complement mechanism shall run and generate the recovered cases for these days.

**Comment by lisphilar** [comment]
I created pull request #317 . @Inglezos , please review it.

## What was changed
- `Scenario` class complements the number of recovered cases automatically with `auto_complement=True` (default) argument and `Scenario.complement()` method.
- Users can restore the raw data with `Scenario.complement_reverse()` method.

## Details of complement
If the number of recovered cases did not change for more than `interval=12` days after reached `max_ignored=100` cases, complement will be applied to the number of recovered cases.

1. Duplicated values will be raplaced with `None`
2. Interpolated with linear function and rounded
3. `Infected` will be re-calculated

Because complement impacts on both of S-R trend analysis and parameter estimation, this will be performed when an instance of `Scenario` class will be created.

**Comment by lisphilar** [comment]
Before (Greece):
![grc_records](https://user-images.githubusercontent.com/7270139/98461739-1438c880-21f2-11eb-908d-208d5b050ed8.jpg)
![grc_trend](https://user-images.githubusercontent.com/7270139/98461741-19961300-21f2-11eb-8c7b-923a68e99aab.jpg)


After:
```Python
snl = cs.Scenario(jhu_data, population_data, "Greece")
snl.records()
snl.trend()
# Resotore the raw data if necessary
# snl.complement_reverse().records()
```

![grc_records](https://user-images.githubusercontent.com/7270139/98461773-56620a00-21f2-11eb-8bc6-3188ff7c7779.jpg)
![grc_trend](https://user-images.githubusercontent.com/7270139/98461775-595cfa80-21f2-11eb-8b7c-c1d41854ad7d.jpg)

The output of `snl.estimate(cs.SIRF); snl.simulate()` is the next figure. Accuracy of 0th phase is low, but this is another issue. #270

**Comment by lisphilar** [comment]
Update:
When complemented, title of the figure of `Scenario.records()` will be changed from "Greece: Cases over time" to "Greece: Cases over time (complemented)" as shown in the previous comment.

**Comment by Inglezos** [comment]
Great, this is the desired handling, thank you! I approved the changes, please merge them to master.

**Comment by lisphilar** [comment]
I merged #317 . This closes #309 and this issue.
